amd_openvx_extensions/amd_rpp/source/tensor/TensorAddTensor.cpp

```diff
@@ -0,0 +1,248 @@
+    int *srcDims2;
+    RpptROI* pSrcRoi;
+    RpptROI* pDstRoi;
+    size_t inputTensorDims1[RPP_MAX_TENSOR_DIMS];
+    size_t inputTensorDims2[RPP_MAX_TENSOR_DIMS];
+    vx_enum inTensorType;
+    vx_enum outTensorType;
+};
+
+void update_destination_roi(const vx_reference *parameters, TensorAddTensorLocalData *data, void* roi_tensor_ptr_src, void* roi_tensor_ptr_dst) {
+    data->pDstRoi = (RpptROI *)roi_tensor_ptr_dst;
+    data->pSrcRoi = (RpptROI *)roi_tensor_ptr_src;
+    for (uint i = 0; i < data->nbatchSize; i++) {
+        data->pDstRoi[i].xywhROI.xy.x = data->pSrcRoi[i].xywhROI.xy.x;
+        data->pDstRoi[i].xywhROI.xy.y = data->pSrcRoi[i].xywhROI.xy.y;
+    }
+}
+
+static vx_status VX_CALLBACK refreshTensorAddTensor(vx_node node, const vx_reference *parameters, vx_uint32 num, TensorAddTensorLocalData *data) {
+    vx_status status = VX_SUCCESS;
+
+    void *roi_tensor_ptr_src, *roi_tensor_ptr_dst;
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_OPENCL
+        return VX_ERROR_NOT_IMPLEMENTED;
+#elif ENABLE_HIP
+         return VX_ERROR_NOT_IMPLEMENTED;
+#endif
+    }
+    else if (data->deviceType == AGO_TARGET_AFFINITY_CPU) {
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HOST, &data->pSrc1, sizeof(data->pSrc1)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HOST, &data->pSrc2, sizeof(data->pSrc2)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HOST, &data->pDst, sizeof(data->pDst)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HOST, &roi_tensor_ptr_src, sizeof(roi_tensor_ptr_src)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_BUFFER_HOST, &roi_tensor_ptr_dst, sizeof(roi_tensor_ptr_dst)));
+        update_destination_roi(parameters, data, roi_tensor_ptr_src, roi_tensor_ptr_dst);
+    }
+    return status;
+}
+
+static vx_status VX_CALLBACK validateTensorAddTensor(vx_node node, const vx_reference parameters[], vx_uint32 num, vx_meta_format metas[]) {
+    vx_status status = VX_SUCCESS;
+    vx_enum scalar_type;
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[5], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_UINT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #4 type=%d (must be size)\n", scalar_type);
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[6], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_UINT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #5 type=%d (must be size)\n", scalar_type);
+
+    // Check for output parameters
+    vx_tensor output;
+    vx_parameter output_param;
+    size_t num_tensor_dims;
+    vx_uint8 tensor_fixed_point_position;
+    size_t tensor_dims[RPP_MAX_TENSOR_DIMS];
+    vx_enum tensor_type;
+    output_param = vxGetParameterByIndex(node, 2);
```

Comment:
Change this validation related code as per the latest validation function changes in other openvx nodes

Reply:
Done